src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/_params.py

```diff
@@ -0,0 +1,151 @@
+
+    with self.argument_context('ams account storage') as c:
+        c.argument('account_name', account_name_arg_type,
+                   help='The name of the Azure Media Services account within the resource group.')
+        c.argument('storage_account', name_arg_type,
+                   help='The name or resource ID of the secondary storage account to detach from the Azure Media Services account.',
+                   validator=storage_account_id)
+
+    with self.argument_context('ams account sp') as c:
+        c.argument('account_name', account_name_arg_type,
+                   help='The name of the Azure Media Services account within the resource group.')
+        c.argument('sp_name', name_arg_type,
+                   help="The app name or app URI to associate the RBAC with. If not present, a default name like '{amsaccountname}-access-sp' will be generated.")
+        c.argument('sp_password', password_arg_type,
+                   help="The password used to log in. Also known as 'Client Secret'. If not present, a random secret will be generated.")
+        c.argument('role', completer=get_role_definition_name_completion_list)
+        c.argument('xml', help='Enables xml output format.')
+        c.argument('years', type=int, default=None)
+
+    with self.argument_context('ams transform') as c:
+        c.argument('account_name', account_name_arg_type, id_part='name',
+                   help='The name of the Azure Media Services account within the resource group.')
+        c.argument('transform_name', name_arg_type, id_part='child_name_1',
+                   help='The name of the transform.')
+        c.argument('preset_names', arg_type=get_enum_type(EncoderNamedPreset),
+                   nargs='+',
+                   help='Space-separated list of built preset names.')
+        c.argument('description', help='Customer supplied description of the transform.')
+
+    with self.argument_context('ams transform list') as c:
+        c.argument('account_name', id_part=None)
+
+    with self.argument_context('ams asset') as c:
+        c.argument('account_name', account_name_arg_type, id_part='name',
+                   help='The name of the Azure Media Services account within the resource group.')
+        c.argument('asset_name', name_arg_type, id_part='child_name_1',
+                   help='The name of the asset.')
+
+    with self.argument_context('ams asset list') as c:
+        c.argument('account_name', id_part=None)
+
+    with self.argument_context('ams asset create') as c:
+        c.argument('alternate_id', help='The alternate id of the asset.')
+        c.argument('description', help='The asset description.')
+        c.argument('asset_name', name_arg_type, help='The name of the asset.')
+
+    with self.argument_context('ams asset get-sas-urls') as c:
+        c.argument('permissions', arg_type=get_enum_type(AssetContainerPermission),
+                   help='The permissions to set on the SAS URL.')
```

Comment:
Does this accept multiple permissions?  If so, it should accept `nargs='+'`

Reply:
This argument accepts multiple permissions so we added `nargs='+'`.